core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -10171,17 +10164,26 @@ matched clause of merge). If the upsert caused a row to be updated in the
 base table then the old version of the row will have to be deleted from 
 indexes, and a new version inserted. Upsert is being transformed to merge
 so that we can delete the old version of an updated row from the index.
+
+Upsert is also converted into merge when there are omitted cols with default values and 
+TRAF_UPSERT_WITH_INSERT_DEFAULT_SEMANTICS is set to  OFF in case of aligned format table or 
+omitted current timestamp cols in case of non-aligned row format
```

Comment:
Can you explain this again, I would have thought it's the other way round: With non-aligned format, if we omit any column then we run a chance of seeing the value of an older row. Is that a "feature"? If so and if TRAF_UPSERT_WITH_INSERT_DEFAULT_SEMANTICS is set to ON, will that cause a difference between the behavior with aligned and non-aligned format? My apologies, I think you explained this to me before.